### PR TITLE
Lift update binding logic from DuckTableEntry to TableCatalogEntry

### PR DESCRIPTION
**Pull request by bleskes**
Binding the update operator requires some careful handling of indices. All of that currently happens in the DuckTableEntry class but it seems there's nothing really DuckTable specific in it. We can lift it to the base class TableCatalogEntry, so the logic is available for custom catalog implementation.

### CODE REVIEW

**Comment by Mytherin** [comment]
Thanks for the PR! LGTM. This might require some tweaking in the SQLite extension as well. The single-file compilation (compiling with `DISABLE_UNITY=1`) is also failing in the CI - likely due to some missing includes. Could you have a look at that?

**Comment by bleskes** [comment]
@Mytherin the compilation issue should be fixed now. Any tips on how to the SQLite extension? I'm not sure what the process is because in another repo

**Comment by Mytherin** [comment]
Generally we send a PR and merge it into the other repo first, and then update extensions.csv with the new git hash as part of the PR

**Comment by bleskes** [comment]
@Mytherin I've put up a pr to sql lite, but I'm clearly not following the right process to validate it. Can you point me at where I can read the steps?

**Comment by Mytherin** [comment]
The SQLite scanner still uses the old extension mechanism which is not very well documented. You can update `extensions.csv` in this PR (and run the format fix) in order to move the SQLite version forward.

**Comment by bleskes** [comment]
@Mytherin done. Thanks.

**Comment by bleskes** [comment]
@Mytherin lmk if the thread santinzer issue looks to you like it was caused by this pr.

File: src/include/duckdb/catalog/catalog_entry/table_catalog_entry.hpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ class TableFunction;
 struct FunctionData;
 
 class TableColumnInfo;
-class TableIndexInfo;
+class ColumnSegmentInfo;
```
Comment (carlopi) [inline src/include/duckdb/catalog/catalog_entry/table_catalog_entry.hpp:40]:
```suggestion
struct ColumnSegmentInfo;
```

File: src/include/duckdb/storage/table/row_group.hpp [hunk 2]
```diff
@@ -31,7 +31,7 @@ class RowGroupCollection;
 class RowGroupWriter;
 class UpdateSegment;
 class TableStatistics;
-class TableStorageInfo;
+class ColumnSegmentInfo;
```
Comment (carlopi) [inline src/include/duckdb/storage/table/row_group.hpp:34]:
```suggestion
struct ColumnSegmentInfo;
```

File: src/include/duckdb/storage/table/row_group_collection.hpp [hunk 3]
```diff
@@ -26,6 +26,7 @@ struct TableAppendState;
 class DuckTransaction;
 class BoundConstraint;
 class RowGroupSegmentTree;
+class ColumnSegmentInfo;
```
Comment (carlopi) [inline src/include/duckdb/storage/table/row_group_collection.hpp:29]:
```suggestion
struct ColumnSegmentInfo;
```

**Review by carlopi — COMMENTED** [review]
+ I would add line:
```
catalog/catalog_entry/table_catalog_entry.cpp 3
```
somewhere in .github/config/uncovered_lines.csv.

Thanks!

**Pull request merged**